yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,162 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    float_or_none,
+    int_or_none,
+    traverse_obj,
+)
+
+
+class AcFunVideoIE(InfoExtractor):
+    IE_NAME = 'AcFunVideoIE'
+    _VALID_URL = r'(?x)https?://(?:www\.acfun\.cn/v/)ac(?P<id>[_\d]+)'
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'info_dict': {
+            'id': '35283657',
+            'title': '【十五周年庆】文章区UP主祝AcFun生日快乐！',
+            'duration': 455.21,
+            'timestamp': 1655289827,
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+        'skip': 'Geo-restricted to China',
+    }, {
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'only_matching': True,
+    }]
+
+    @staticmethod
+    def parse_format(info):
+        vcodec, acodec = None, None
+
+        codec_str = info.get('codecs') or ''
+        m = re.match('(?P<vc>[^,]+),(?P<ac>[^,]+)', codec_str)
+        if m:
+            vcodec = m.group("vc")
+            acodec = m.group("ac")
+
+        return {
+            'url': info.get('url'),
+            'fps': int_or_none(info.get('frameRate')),
+            'width': int_or_none(info.get('width')),
+            'height': int_or_none(info.get('height')),
+            'vcodec': vcodec,
+            'acodec': acodec,
+            'tbr': float_or_none(info.get('avgBitrate'))
+        }
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        video_id = mobj.group('id')
+
+        webpage = self._download_webpage(url, video_id)
+        json_all = self._search_json(r'window.videoInfo\s*=\s*', webpage, 'videoInfo', video_id)
+
+        video_info = json_all['currentVideoInfo']
+        playjson = self._parse_json(video_info['ksPlayJson'], video_id)
+        video_inner_id = traverse_obj(json_all, ('currentVideoInfo', 'id'))
+
+        ext = determine_ext(video_info.get('fileName'), 'mp4')
```

Comment:
Is this possible to generate `m3u8` ? if yes, i think it's better to hardcoding the ext imo

Reply:
yes, the url in format is m3u8, like
`https://tx-safety-video.acfun.cn/mediacloud/acfun/acfun_video/hls/tBeJ6NxIf7xlrBTdgaS37IByKhPByahiA9tc4Lln29pSSUUV52oY_ogylZNgZj0F.m3u8?pkey=ABD2fS0rMWHIF544VBp_IVO2_4u_g7FLPp8etV6L6pw1ivbbYOkW7sJVSQGCSB3H1nXYUDnO8rdsUQFxvcvcSVMTbbGRhFQX2Oa7qlSxhKTlDLAO5oeWqHl2Cy4cs6aOkUaCHRynPoh6n2iHUeCHpTG3r-lKO_rBDWa2KZtcXMhj7-0Jd3k-LNCscNJQfBTHik8SU0ohpJ0N3pdLSUzshN7q2BpgSfOQKZs2lZ5F1hzfQA&safety_id=AAJKHjYqdq9L8aKcTVDyj9n0`

I dont know if all m3u8 should set to mp4?

Reply:
You can hardcoding it to mp4 ( actually you can use other extension but mp4 is most popular one), i think m3u8 is not something user expect imo

Reply:
Do hardcode `mp4` if it's m3u8

Reply:
> Do hardcode `mp4` if it's m3u8. 
ok

> Also, use `_extract_m3u8_formats_and_subtitles` to get formats instead of constructing dict manually
ok, thanks